src/pybind/mgr/volumes/fs/operations/resolver.py

```diff
@@ -21,9 +21,20 @@ def resolve(vol_spec, path):
 
 def resolve_trash(vol_spec, path):
     parts = splitall(path)
-    if len(parts) != 6 or os.path.join(parts[0], parts[1]) != vol_spec.subvolume_prefix or \
-       parts[4] != '.trash':
+
+    if len(parts) == 6:
```

Comment:
Maybe add some comment like the following:
This is to resolve the trashpath of subvolumes that retain snapshots.  The subvolume trash path has the following syntax 
/\<vol_spec.subvolume_prefix\>/\<groupname\>/\<subvolname\>/.trash/\<subvolUUID\> ?